pkg/config/cdn.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+import "net/url"
+
+// CDNConfig specifies the properties required to use a CDN as the storage backend
+type CDNConfig struct {
+	Endpoint string `envconfig:"CDN_ENDPOINT"`
+	Timeout  int    `validate:"required"`
+}
+
+// CDNEndpointWithDefault returns CDN endpoint if set
+// if not it should default to clouds default blob storage endpoint e.g
+func (c *CDNConfig) CDNEndpointWithDefault(value *url.URL) *url.URL {
```

Comment:
Do we need this function? Can we not just work with the URL as a string and not a *ur.URL?

Reply:
I've left this in for the `WithDefault` behavior.

Currently the default passed in is the base URI of the storage implementation being used. We could add logic here to default to the Endpoint/URI of the storage config defined in `StorageType`. That would require this config package to be aware of a little more business logic, and if we're doing that we could also just move all the logic in the `GetStorage()` method in `actions/storage.go` to this package as well.